csp/csp.go

```diff
@@ -260,14 +260,40 @@ func (sp *StoragePlugin) Serve(ctx context.Context, lis net.Listener) error {
 		sp.server = grpc.NewServer(sp.ServerOpts...)
 
 		// Register the CSI services.
-		if s := sp.Controller; s != nil {
-			csi.RegisterControllerServer(sp.server, s)
-		}
+		// Always register the Identity service
 		if s := sp.Identity; s != nil {
 			csi.RegisterIdentityServer(sp.server, s)
+			log.Debug("Registered Identity Service")
+		}
+
+		// determine which of the controller/node services to register
+		nsvc := true
+		csvc := true
+		if s, ok := gocsi.LookupEnv(ctx, EnvVarNodeSvcOnly); ok {
+			if b, err := strconv.ParseBool(s); err != nil {
```

Comment:
You got pulled into the pattern problem. I do this too, so often, when parsing for bool. It should be `err == nil`